pkg/cmd/cli/cmd/set/imagelookup.go

```diff
@@ -0,0 +1,235 @@
+
+	PrintTable  bool
+	PrintObject func(runtime.Object) error
+
+	Enabled bool
+}
+
+// NewCmdImageLookup implements the set image-lookup command
+func NewCmdImageLookup(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &ImageLookupOptions{
+		Out:     out,
+		Err:     errOut,
+		Enabled: true,
+	}
+	cmd := &cobra.Command{
+		Use:     "image-lookup STREAMNAME [...]",
+		Short:   "Use an image stream when short image names are provided",
+		Long:    fmt.Sprintf(imageLookupLong, fullName),
+		Example: fmt.Sprintf(imageLookupExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			kcmdutil.CheckErr(options.Run())
+		},
+	}
+
+	kcmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter on")
+	cmd.Flags().BoolVar(&options.All, "all", options.All, "If true, select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename, directory, or URL to file to use to edit the resource.")
+
+	cmd.Flags().BoolVar(&options.Enabled, "enabled", options.Enabled, "Mark the image stream as resolving tagged images in this namespace")
+
+	cmd.MarkFlagFilename("filename", "yaml", "yml", "json")
+
+	return cmd
+}
+
+// Complete takes command line information to fill out ImageLookupOptions or returns an error.
+func (o *ImageLookupOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string) error {
+	cmdNamespace, explicit, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	clientConfig, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	o.OutputVersion, err = kcmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
+	if err != nil {
+		return err
+	}
+
+	o.PrintTable = len(args) == 0 && !o.All
+
+	mapper, typer := f.Object()
+	o.Builder = resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(explicit, &resource.FilenameOptions{Recursive: false, Filenames: o.Filenames}).
+		SelectorParam(o.Selector).
+		SelectAllParam(o.All).
+		ResourceNames("imagestreams", args...).
+		Flatten()
+	if len(args) == 0 {
+		o.Builder.SelectAllParam(true)
+		o.Builder.ResourceTypes("imagestreams")
+	}
+
+	output := kcmdutil.GetFlagString(cmd, "output")
+	if len(output) != 0 {
+		o.PrintObject = func(obj runtime.Object) error { return f.PrintObject(cmd, mapper, obj, o.Out) }
+	}
+
+	o.Encoder = f.JSONEncoder()
+	o.ShortOutput = kcmdutil.GetFlagString(cmd, "output") == "name"
+	o.Mapper = mapper
+
+	return nil
+}
+
+// Validate verifies the provided options are valid or returns an error.
+func (o *ImageLookupOptions) Validate() error {
```

Comment:
Sigh... Having empty method just to follow pattern is not what I'd encourage. If it's empty - drop it. If needed add it following patterns.